File: t-tasks/src/main/java/com/teo/ttasks/injection/module/fragment/TasksFragmentModule.java
package com.teo.ttasks.injection.module.fragment;

import com.teo.ttasks.data.local.PrefHelper;
import com.teo.ttasks.data.remote.TasksHelper;
import com.teo.ttasks.ui.fragments.tasks.TasksPresenter;

import dagger.Module;
import dagger.Provides;

@Module
public abstract class TasksFragmentModule {

    @Provides
    static TasksPresenter provideTasksPresenter(TasksHelper tasksHelper, PrefHelper prefHelper) {
        return new TasksPresenter(tasksHelper, prefHelper);
    }
}

File: t-tasks/src/main/java/com/teo/ttasks/injection/module/fragment/TasksFragmentModule.kt
package com.teo.ttasks.injection.module.fragment

import com.teo.ttasks.data.local.PrefHelper
import com.teo.ttasks.data.remote.TasksHelper
import com.teo.ttasks.ui.fragments.tasks.TasksPresenter

import dagger.Module
import dagger.Provides

@Module
class TasksFragmentModule {

    @Provides
    internal fun provideTasksPresenter(tasksHelper: TasksHelper, prefHelper: PrefHelper): TasksPresenter {
        return TasksPresenter(tasksHelper, prefHelper)
    }
}
